lib/Parser/Scan.cpp

```diff
@@ -2087,36 +2093,11 @@ tokens Scanner<EncodingPolicy>::ScanCore(bool identifyKwds)
             case '-':
                 p++;
                 token = tkDec;
-                if (m_fHtmlComments)
+                if (!m_fIsModuleCode)
                 {
-                    int i = 0;
-                    while ('-' == PeekFirst(p + i, last)) //Have already seen --, skip any further - characters
-                        i++;
-                    if ('>' == PeekFirst(p + i++, last)) //This means we've got a --------------------------->.
```

Comment:
So the `-------->` is no longer handled, correct?

There should be a test verifying this.

Also, what do the other browsers do about this case?  If they also allow `------>` then we should probably disregard the spec and maintain this behavior for compatibility.